src/components/selection_input/selection_input.ts

```diff
@@ -245,19 +241,74 @@ export class SelectionInput extends Component<Props, SpreadsheetChildEnv> {
     this.props.onSelectionConfirmed?.();
   }
 
-  disable() {
+  confirm() {
     this.env.model.dispatch("UNFOCUS_SELECTION_INPUT");
-    const ranges = this.env.model.getters.getSelectionInputValue(this.id);
+    const ranges = this.env.model.getters
+      .getSelectionInput(this.id)
+      .filter((r) => this.env.model.getters.isRangeValid(r.xc));
     if (this.props.required && ranges.length === 0) {
       this.state.isMissing = true;
     }
+    for (const range of ranges) {
+      const zone = toUnboundedZone(range.xc);
+      if (zone.bottom !== zone.top && zone.left != zone.right) {
+        if (zone.right) {
+          //test if we don't have something like 1:2
+          this.env.model.dispatch("CHANGE_RANGE", {
+            id: this.id,
+            rangeId: range.id,
+            value: zoneToXc({
+              left: zone.left,
+              right: zone.left,
+              top: zone.top,
+              bottom: zone.bottom,
+            }),
+          });
+          for (var j = zone.left + 1; j <= zone.right; ++j) {
```

Comment:
Splitting the zone on columns is business logic related to charts.
Nothing tells *every use* of the `<SelectionInput/>` component has the same logic.
It doesn't belong here.

It should probably be something updated with the component `"range"` prop.
I don't know why its value is defined only at `setup` in the config panel component and why they aren't dynamic